hypothesis-python/tests/cover/test_type_lookup.py

```diff
@@ -224,6 +225,14 @@ def __init__(self, arg: T) -> None:
         self.arg = arg
 
 
+class Lines(Sequence[str]):
```

Comment:
Perhaps you also want to test against a generic class with multiple parametric types (*e.g.*, `Mapping[int, int]`) so that you have covered kind of an "induction"?

Here's a simplified example from https://github.com/mristin/python-by-contract-corpus/blob/f2d43b6f138a75a01c5709a604da87662ac83392/correct_programs/aoc2020/day_10_adapter_array.py#L46 (I removed the inheritance from `Mapping` due to the issue #2951).

```python
from typing import Mapping, Iterator, cast


class HistogramOfDeltas(Mapping[int, int]):
    def __new__(cls, mapping: Mapping[int, int]) -> "HistogramOfDeltas":
        assert all(delta > 0 for delta in mapping.keys())
        assert all(count >= 0 for count in mapping.values())
        return cast(HistogramOfDeltas, mapping)

    def __getitem__(self, key: int) -> int:
        raise NotImplementedError("Only for type annotations")

    def __iter__(self) -> Iterator[int]:
        raise NotImplementedError("Only for type annotations")

    def __len__(self) -> int:
        raise NotImplementedError("Only for type annotations")
```